### PR TITLE
[INTEL MKL] Add MKL-DNN quantized Matmul op with some fusions - Part3.

### PR DESCRIPTION
**Pull request by mdfaijul**
This PR is to replace an older PR #26271 by splitting into three. Current PR is part3. It adds unit tests for MKL-DNN quantized Matmul ops. Tests depends on part1 (#26909) and part2 (#26910).

### CODE REVIEW

**Review by penpornk — COMMENTED** [review]
Just a note that we will look at this once we are done with its dependencies (parts 1 and 2).

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]: Please move this comment to directly above `ConvMklToTF`.

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]: Per our [Style Guide](https://google.github.io/styleguide/cppguide.html#Constant_Names) for constant variables,
```suggestion
static const uint8 kDummyTensor[] = {0, 0, 0, 0, 0, 0, 0, 0};
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
static const TensorShape kDummyShape({2, 4});
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
          .Input(FakeInput(DT_UINT8))  // MKL second tensor
```
Please fix other places too.

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
  // Final result after Bias addition:
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
  // Final result after Bias addition:
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
// Two small matrices A of type uint8 (converted from signed integer)
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
  // quantized values as Au8 = (Af32 - Min(A𝑓32)) * Qa, where
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
  // First calculate C = A * B,
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
  // Qa = 255/(Max(A𝑓32)–Min(A𝑓32)). For example, -1 is represented
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]:
```suggestion
// Simple test for Matrix multiplication with Bias, Relu and
```

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]: Just curious, is 256 (not 255) intentional?

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]: Please add a newline before this line. Also,
```suggestion
#endif  // INTEL_MKL
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! I appreciate the detailed comments. I have minor formatting comments.

**Comment by mdfaijul** [comment]
@penpornk Thank you very much for the comments. Please check the latest commit.

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op_test.cc]: Missing one closing parenthesis.
```suggestion
  // quantized values as Au8 = (Af32 - Min(A𝑓32)) * Qa, where
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for your quick fixes! One more change and this should be good to go too.

File: tensorflow/core/kernels/mkl_qmatmul_op_test.cc
Comment (mdfaijul) [reply tensorflow/core/kernels/mkl_qmatmul_op_test.cc]: @penpornk Thank you very much. Please check the latest commit.

**Review by penpornk — COMMENTED** [review]
Thank you! I'll approve this when parts 2 and 3 are merged. (If I approve it now the tests will just fail.)

**Review by penpornk — APPROVED** [review]
Approving this since part 1 (https://github.com/tensorflow/tensorflow/pull/26909) has been merged.

**Comment by penpornk** [comment]
@mdfaijul Some of the failed tests are from the missing `mkl_qmatmul_op` target. (My bad, I thought the build is also guarded by `is_mkl`. I'll look into that after all three parts are merged.) We still have some time. I'll wait until part 2 is merged and rerun the tests then.

**Pull request merged**
